set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java

```diff
@@ -166,11 +166,15 @@ public PersistentUserSet findByID(String id) {
 	}
 
 	@Override
-	public long getDistinctCreators() {
-		// create query : { type: { $eq: Collection } }
-		DBObject match = new BasicDBObject(WebUserSetFields.TYPE,
-				new BasicDBObject(UserSetMongoConstants.MONGO_EQUALS, UserSetTypes.COLLECTION.getJsonValue()));
-		return getDao().getCollection().distinct(WebUserSetFields.CREATOR, match).size();
+	public long getDistinctCreators(String type) {
+		// create query : { type: { $eq: <type> } }
+		if (type != null) {
+			DBObject match = new BasicDBObject(WebUserSetFields.TYPE,
+					new BasicDBObject(UserSetMongoConstants.MONGO_EQUALS, type));
+			return getDao().getCollection().distinct(WebUserSetFields.CREATOR, match).size();
+		} else {
+			return getDao().getCollection().distinct(WebUserSetFields.CREATOR).size();
```

Comment:
this call still fetches the whole results in the memory. the count() method should be used instead getDao().count(Query).  (it should be looked in debugger that the creator list is not fetched from the database, otherwise limit=0 can be set in the query)